napi.h

```diff
@@ -1820,6 +1821,193 @@ namespace Napi {
+
+    // Default set + Resource + Context
+    template <typename ResourceString, typename ContextType>
+    static ThreadSafeFunction New(napi_env env,
+                                  const Function& callback,
+                                  const Object& resource,
+                                  ResourceString resourceName,
+                                  size_t maxQueueSize,
+                                  size_t initialThreadCount,
+                                  ContextType* context);
+
+    // Default set + Resource + Finalizer
+    template <typename ResourceString, typename Finalizer>
+    static ThreadSafeFunction New(napi_env env,
+                                  const Function& callback,
+                                  const Object& resource,
+                                  ResourceString resourceName,
+                                  size_t maxQueueSize,
+                                  size_t initialThreadCount,
+                                  Finalizer finalizeCallback);
+
+    // Default set + Resource + Finalizer + Data
+    template <typename ResourceString, typename Finalizer,
+              typename FinalizerDataType>
+    static ThreadSafeFunction New(napi_env env,
+                                  const Function& callback,
+                                  const Object& resource,
+                                  ResourceString resourceName,
+                                  size_t maxQueueSize,
+                                  size_t initialThreadCount,
+                                  Finalizer finalizeCallback,
+                                  FinalizerDataType* data);
+
+    // Default set + Resource + Context + Finalizer
+    template <typename ResourceString, typename ContextType, typename Finalizer>
+    static ThreadSafeFunction New(napi_env env,
+                                  const Function& callback,
+                                  const Object& resource,
+                                  ResourceString resourceName,
+                                  size_t maxQueueSize,
+                                  size_t initialThreadCount,
+                                  ContextType* context,
+                                  Finalizer finalizeCallback);
+
+    // Default set + Resource + Context + Finalizer + Data
+    template <typename ResourceString, typename ContextType,
+              typename Finalizer, typename FinalizerDataType>
+    static ThreadSafeFunction New(napi_env env,
+                                  const Function& callback,
+                                  const Object& resource,
+                                  ResourceString resourceName,
+                                  size_t maxQueueSize,
+                                  size_t initialThreadCount,
+                                  ContextType* context,
+                                  Finalizer finalizeCallback,
+                                  FinalizerDataType* data);
```

Comment:
For consistency we should change `napi_env env` to `Napi::Env env` in the factory functions above.

Reply:
Could you let me know more details about the consistency you mentioned?
I'm not sure but the most of New() methods are already using `napi_env` instead of `Napi::Env`.

FYI, please see the following `grep` result.
```
$ grep -Enr "New\((Napi::Env|Env|napi_env)" napi.h

693:    static External New(napi_env env, T* data);
697:    static External New(napi_env env,
702:    static External New(napi_env env,
715:    static Array New(napi_env env);
716:    static Array New(napi_env env, size_t length);
908:    static DataView New(napi_env env,
910:    static DataView New(napi_env env,
913:    static DataView New(napi_env env,
961:    static Function New(napi_env env,
968:    static Function New(napi_env env,
1005:      static Deferred New(napi_env env);
1026:    static Buffer<T> New(napi_env env, size_t length);
1027:    static Buffer<T> New(napi_env env, T* data, size_t length);
1031:    static Buffer<T> New(napi_env env, T* data,
1036:    static Buffer<T> New(napi_env env, T* data,
1285:    static Error New(napi_env env);
1286:    static Error New(napi_env env, const char* message);
1287:    static Error New(napi_env env, const std::string& message);
1312:    static TError New(napi_env env,
1324:    static TypeError New(napi_env env, const char* message);
1325:    static TypeError New(napi_env env, const std::string& message);
1333:    static RangeError New(napi_env env, const char* message);
1334:    static RangeError New(napi_env env, const std::string& message);
1829:    static ThreadSafeFunction New(napi_env env,
1837:    static ThreadSafeFunction New(napi_env env,
1846:    static ThreadSafeFunction New(napi_env env,
1856:    static ThreadSafeFunction New(napi_env env,
1866:    static ThreadSafeFunction New(napi_env env,
1877:    static ThreadSafeFunction New(napi_env env,
1888:    static ThreadSafeFunction New(napi_env env,
1897:    static ThreadSafeFunction New(napi_env env,
1907:    static ThreadSafeF1897:    static ThreadSafeFunction New(napi_env env,
1907:    static ThreadSafeFunction New(napi_env env,
1918:    static ThreadSafeFunction New(napi_env env,
1929:    static ThreadSafeFunction New(napi_env env,
1941:    static ThreadSafeFunction New(napi_env env,
2003:    static ThreadSafeFunction New(napi_env env,
```